### ISSUE TITLE
After Burner II has background color issues

### ISSUE THREAD

**Issue by Wilstorm** [libretro/mame2003-plus-libretro#319]
After Burner 2 (aburner2.zip) has some color issues and also left/right edge overlap. I am not sure how to describe it but it's easy to spot in the pics.

**Incorrect:**
![1532716847719-aburner-180727-192858-resized](https://user-images.githubusercontent.com/8192258/43347778-6fdd6dd0-91bc-11e8-8976-39093d1d0a30.png)

**Correct:**
![1532716899221-aburner-180727-193057-resized](https://user-images.githubusercontent.com/8192258/43347785-77b36082-91bc-11e8-8bc7-88fd15568545.png)

**Comment by grant2258** [comment]
this isint happening to me on windows or retropie the overlap is there but its black might be a shader or something

**Comment by Wilstorm** [comment]
Yeah it's Afterburner II and you have to let it loop the intro's for a few minutes until the pilot image flashes across the screen for a second. I did disable the shader thinking that sounds reasonable as a cause but it shows the same issue either way.

**Comment by grant2258** [comment]
i checked in mame2003 its an official bug so at least its not something new maybe arcadez will know more about this. I know there was a system16 update that was reverted

**Comment by arcadez** [comment]
Alotta the Sega games using the System 16 18 etc etc hardware do have graphical niggles in this core After Burner II being one of them, it's not until the MAME88 - 91 dev cycle when there was a major graphical update that these niggles were sorted allbeit with a performance drop for some of them.

BTW there are more niggles with After Burner II that just on the title screen a bonus point
will be awarded to anyone that spots em :)

**Comment by Wilstorm** [comment]
@arcadez - Thanks guys for looking into it. Stuart over on the RetroPie forums pointed it out and asked if there was a fix so I created an issue. I play it thinking everything looks good so I would definitely miss the bonus point(s). Shoot I completely missed the shadows in Golden Axe II also. Nostalgia trumps! :)

**Comment by arcadez** [comment]
@Wilstorm There's usually always a fix available from later MAME cores the problem is as always will the code be compatable with this older core and the workload required may make it not worth the hastle for a minor improvement.

In the case of After Burner II it fails on both of the above

**Comment by Wilstorm** [comment]
@arcadez - Yes, double fails, definitely I know what you're saying and I agree the issue is pretty minor and the game fully playable. In fact I didn't even notice it. It does work for him in mame2010 (0.139) which is above the minimum version where they did the graphical overhaul so it makes sense why 2010 works. 

Oh and the trivia/informational links you throw into posts are fun and I appreciate them. It lightens things up and I've learned some new interesting things also. With that thought on how this issue came about I was wondering if you guys had a preference on game issues.

@grant2258 @markwkidd @arcadez - It seems there's definitely more people using Plus and of course more users means more bugs until they get ironed out. On Friday 3 bugs popped up and I kind of directed them over here but then I wasn't sure if you prefer to see issues on the forum first. One guy was looking at the After Burner II graphic glitches, another guy was working on getting CHD's up and running (I did some testing and they work fine so he's looking on his end) and a 3rd is having an issue with Revolution X with black screens on the selection and score screen. 

Basically I was looking for direction from you guys. Would you rather have them post issues on the forums first or can they be directed here to open new issues? I open an issue for After Burner but it didn't seem to make much sense to take their issue, retype it, and open a new issue here vs. directing them here to open their own issue. 

I think as Plus gains popularity there will be more issues until things hit the bell curve, level out and eventually slow down (then you'll have nothing left to do! ;) I thought it might help to have some direction on how you would want to handle new issues. I wanted to check with you guys before suggesting going here for new issues. I know Grant has been addressing most of them on the forums and it seems to be working ok too. Any thoughts?

**Comment by grant2258** [comment]
its neither here nor there where they report it. It would get more attention here though. I peronally havent used chds and dont have any. What I can tell you is chds go in the roms directory.

chd diffs however go into options.libretro_save_path on retropie this will be romsdir/mame2003-plus on ra its will be in savefile path. I believe system (bios) dir would have been a better place for this though but it is what it is.

**Comment by Wilstorm** [comment]
@grant2258 - Thanks, I like the go with the flow. I think that's why I like it here because you guys don't have a bunch of hard and fast rules for everything but if you decide you need to rework a flow I can work with that too.

I always dump my CHDs in with the ROMs but in a folder of the ROM name. I tested several last week on RetroPie and they work ok but entirely to slow on the Pi. I wouldn't mind seeing some of those playable working at some point like San Francisco Rush. Racing games are appealing to me lately. We used to play a lot of Burnout on the XBOX.

**Comment by grant2258** [comment]
revolutionx im guessing its setting the lightgun inputs. Since my last conversation about input matter got silenced and closed Mark can deal with that part as he said its withing his responsibility. Only lightgun1 is mapped atm (rstick on controller1) 3 players would need setup

**Comment by markwkidd** [comment]
If the lightgun input mode is set, the abilities for lightgun games are
more limited than if the lightgun is set in retro arch as a mouse device.
IIRC. Lightguns set in RA as generic analog input devices may also work
better than actually using the lightgun setting.

**Comment by Wilstorm** [comment]
@markwkidd - Here's what Darksavior posted on Revolution X. If I can do any testing if there's something to tweak just let me know.

_Revolution X has a black screen on the scoring section at the end of a level and also a black screen on the level select screen._

**Comment by arcadez** [comment]
The fix for Rev-X was done at the same time as the one which fixed the random end of level
crash in Rampage World Tour

- 9th May 2007: Aaron Giles - Having recently touched all of the TMS34010-based games led me to start looking into some long-standing issues with some of the Williams/Midway Z/Y/X/Wolf-unit games. The first thing to do was to add save state support to them, because playing Revolution X more than once to get to the place where the video craps out is not for the faint of heart. This was relatively straightforward, and also had the side benefit of making some useful data available in the debugger (such as the local_videoram array and the dma_registers array). Side note: any array or pointer you register with state_save_register_global_array/_pointer automatically becomes a viewable option in a memory window in the debugger. This is very handy if you need to expose any internal state. Anyway, it turned out that the Revolution X bug was caused by over-aggressive masking in the blitter code. It was masking out one too many bits in the clipping window and this led to the right edge of the clip window being less than the left edge, effectively clipping out everything during the blit. Hence the black screen.

- 0.114u2: Aaron Giles fixed Revolution X gameplay/finish a level.

I've done the Rampage WT fix for this core i could do the one for Rev-X should the code be 
compatable, testing my end will be tricky however as calibrating the guns on an xbox gamepad
was never my strong point.

**Comment by Wilstorm** [comment]
@arcadez - Is there a tutorial on how to calibrate a xbox gamepad for testing or is that getting in deep? If the code is compatible and it's something I can do I'll give it a whirl.

**Comment by arcadez** [comment]
Dunno TBH i just used to steal the NVRAM and savestates from a rival project :) i never kept
Rev-X in my romlist but doin the above worked a treat for Terminator 2

**Comment by grant2258** [comment]
the game will ask u to calibrate it when u load it

**Comment by Wilstorm** [comment]
lol...so Terminator 2 works now where it didn't previously?

**Comment by arcadez** [comment]
Nah it always worked bigman you just had to calibrate it firstly

**Comment by Wilstorm** [comment]
Damn you move quick when you add a fix though, good stuff! ;)

**Comment by Wilstorm** [comment]
@arcadez - I see what you're saying. I was just messing with RevX and Term2. RevX I couldn't get the controller fully configured (upper left wouldn't respond to pull trigger but lower right was fine).

Term2 I was able to get it fully configured but the gun scope was below the bottom of the screen with bullets flying up and all I could do was arc back and forth without seeing my scope and targeting theeee....TERMIANATORRS! :)

**Comment by arcadez** [comment]
Aye it's no easy hence why i just stole someone else's :) on a totally unrelated matter this
reminds me you have to enter the service mode and switch the control type for Arkanoid Returns
otherwise the paddle wont move.

**Comment by Wilstorm** [comment]
@arcadez - One more quick question. So I found an NVRAM file and it works using the left stick but of course it snaps to center when released and has a small range to work with moving the gun around and shooting makes it tough to play. I can see a mouse working but can it work on gamepad decent or not really and need a mouse/light gun?

Thanks for the tip on Arkanoid Returns. I like to setup my nvram and cfg files ready to dump into folders on a new setup.

**Comment by arcadez** [comment]
@Wilstorm When using MAME on the xbox gamepad we tended to assign the inputs for LG games 
to the D-pad rather than the sticks otherwise the crosshair would move far to quickly.

**Comment by grant2258** [comment]
yea the same happens on analog stick here. I did notice the analog is a lot better on the xbox one controller for some reason when playing analog games maybe just a better resolution for the analog singnal

**Comment by Wilstorm** [comment]
Here's the [nvram files](https://ufile.io/78e7c) for RevolutionX and Terminator 2. I was able to set the upper left to 0 and lower right to 255 (using a mouse) which seems to work fine. I think I have all players configured correctly.

When I start the game with a joystick (Retroarch D-pad to analog) everything is centered and shoots fine but the center of the analog stick is 0 I believe and it goes up, down, left & right to 127 max, so basically with those values and the round hole of the thumbstick you get a perfect diamond firing pattern with the corners unreachable. It's hard to explain but after you mess with configuring a joystick only it makes sense because you can see the values change when moving the thumbstick. The thumbstick throw is tight so that cursor really zips around the screen to fast. 

When I try and configure with joystick only I can't get the cross-hairs on the screen once I start a game (they are always off the bottom of the screen).

When I download a term2.nv RAM file pre-configured it was the same way so they are good I believe unless I need to use the D-Pad and can get full range on the screen with a thumbstick. Anyway they are linked above.

Oh it's not bad on a keyboard for testing. The crosshair seems to "follow" the keyboard and stays in place without snapping back to center when releasing keys so you can actually test ok with it.
